src/plugin/config/configprovider.js

```diff
@@ -114,6 +115,8 @@ plugin.config.Provider.prototype.loadConfig = function(id, config) {
     if (!descriptor) {
       descriptor = new os.data.ConfigDescriptor();
       descriptor.setId(descriptorId);
+      descriptor.setType(config['type']);
```

Comment:
Are these two lines needed? `ConfigDescriptor` will get the type and description from the base configs set by the `setBaseConfig` call below. The `getType` and `getDescription` overrides on `ConfigDescriptor` will search the configs for the appropriate field, so you shouldn't need to call these setters.

`ConfigDescriptor#getType` does use the `layerType` field on the config, so maybe that needs to be changed/added to the layer config you're creating?